### PR TITLE
Add USE_CLCACHE_MODE to have specific build options

### PR DESCRIPTION
**Pull request by KVS85**
Setup CLCACHE specific options for fast compilation.

### CODE REVIEW

**Comment by KVS85** [comment]
oskar PR: https://github.com/arangodb/oskar/pull/131

File: CMakeLists.txt [hunk 1]
```diff
@@ -748,6 +748,13 @@ option(USE_MAINTAINER_MODE
   OFF
 )
 
+if (MSVC)
```
Comment (dothebart) [inline CMakeLists.txt]: please move this to line 239 - maybe the cmake run itself can benefit.

File: CMakeLists.txt [hunk 2]
```diff
@@ -894,6 +901,11 @@ elseif (MSVC)
   set(CMAKE_CXX_FLAGS_RELEASE        "/MT /O2 /Ob2"                      CACHE INTERNAL "C++ release flags")
   set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/MT /Zi /O2 /Ob1"                  CACHE INTERNAL "C++ release with debug info flags")
 
+  if (USE_CLCACHE_MODE)
```
Comment (dothebart) [inline CMakeLists.txt]: shouldn't this also replace all the Zi's above?

Comment (KVS85) [reply CMakeLists.txt]: Yes. It will replace all to Z7 for CLCACHE.

**Pull request merged**